src/corppa/utils/path_utils.py

```diff
@@ -118,6 +118,20 @@ def get_image_relpath(work_id, page_num):
         raise ValueError(f"Unsupported source '{source}'")
 
 
+def page_number(filename: pathlib.Path) -> str:
+    """Extract and return the page number from a :class:`pathlib.Path`
+    file for content from a single page (e.g., image or text). Returns the
+    page number as a string with leading zeros. (Note: logic is curently
+    specific to Gale/ECCO file naming conventions.)"""
+    # NOTE: this logic is currently specific to Gale/ECCO files,
+    # which look like CW0112029406_00180.txt
+
+    # split the file base/stem name by _ and take the last part
+    pagenum = filename.stem.split("_")[-1]
+    # return the number as a string; strip extra trailing zero
+    return pagenum[:-1]  # strip trailing zero
```

Comment:
Is it worth refactoring this now to take a `source` string as input similar to `get_vol_dir` for future source support?

Reply:
Sorry for the red herring. It hadn't occurred to me that the volume identifier could be extracted from image filename, so the source could be inferred (rather than specified as a second input argument)

Reply:
We can revisit in future if that doesn't hold true for other collections.